File: frontend/src/app/models/ChatRoom.ts
import {Users} from "./Users";

export interface ChatRoom{
  id: number;
  name: string;
  type: string;
  description: string;
  createdDate: string;
  lastMessageTimestamp: string;
  members: Users[];
  createdBy: Users;
  isMuted: boolean;
  isPinned: boolean;
  pictureUrl: string;
}

File: backend/src/main/java/hr/tvz/ljubojevic/chatterbox/model/ChatRoom.java
package hr.tvz.ljubojevic.chatterbox.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class ChatRoom {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String type;
    private String description;
    private LocalDateTime createdDate;
    private LocalDateTime lastMessageTimestamp;

    @ManyToMany
    private List<User> members;

    @ManyToOne
    private User createdBy;

    private boolean isMuted;
    private boolean isPinned;

    private String pictureUrl;
}
